HIP/hip-1068.md

```diff
@@ -0,0 +1,528 @@
+    message TransactionSponsorClaim {
+        /**
+         * The account a spender is noting has provided prior willingness through an transaction fee allowance to
+         * sponsor transaction fees
+         */
+        AccountID sponsor_account_id = 1;
+
+        /**
+         * The intended allowance scope a spender is noting an owner will cover. 
+         */
+        AllowanceScope scope = 2;
+    }
+
+    message TransactionBody {
+        ...
+
+        /**
+         * A set of transaction sponsor claims a user is intending to utilize to cover value transfers in this
+         * transaction
+         */
+        repeated TransactionSponsorClaim transaction_sponsor_claim = 53;
+    }
+```
+
+The updated QueryHeader transaction would also contain the sponsored transaction identifier
+```protobuf
+    message QueryHeader {
+        ...
+
+        /**
+         * The account a spender is noting has provided prior willingness through an HBAR allowance to sponsor the
+         * query transaction fees
+         */
+        AccountID sponsor_account_id = 3;
+    }
+```
+
+> Note: A user must specify the sponsor_account_id for the network to consider the validity of the sponsored transaction. 
+If not specified, the network will treat this as a normal transaction in which the submitter is due the transaction
+fees.
+
+### Error Cases
+
+As with normal transaction it’s possible for a sponsored transaction to fail.
+Existing transaction network rules will be applicable. 
+
+A node that encounters an invalid sponsoring account value will respond with a new response code
+
+```protobuf
+    enum ResponseCodeEnum {
+        ...
+
+        INVALID_SPONSOR_ACCOUNT = 352;
+    }
+```
+
+Transaction sponsorship will be complete in nature, a sponsor must be able to cover the full fee.
+In the case a sponsor lacks insufficient balance to cover an account that transaction will fail with a new response
+code.
+
+```protobuf
+    enum ResponseCodeEnum {
+        ...
+        
+        INSUFFICIENT_SPONSOR_ACCOUNT_BALANCE = 353;
+    }
+```
+
+### Transaction Id
+
+Currently the transaction id of any submitted transaction takes the form of `<accountId>@<validStartTime>` format.
+Usually the `accountId` signifies the user who both submitted and paid for the transaction. With the changes from this
+HIP the `accountId` value in the transaction id of a sponsored transaction will refer only to the submitter and no
+longer the payer. The accountId of the payer will be located in the transaction body. 
+
+### System Contract Functions
+
+Similar to [HIP 906](https://hips.hedera.com/hip/hip-906), the system contract interface would need to be updated to
+support the management of allowances with scope by a smart contract.
+
+The following functions are introduced in this HIP
+```solidity
+    interface IHRC1068 {
+        /// extensions of HIP 514
+        function approve(address token, address spender, uint256 amount, AllowanceScope allowanceScope) external
```

Comment:
Definition of AllowanceScope enum in solidity is missing

Reply:
Added